### PR TITLE
Support BigDL for Spark on k8s

### LINKED ISSUE #1679: Support for Kubernetes-native Spark clusters (`k8s://`)
I would like to be able to submit jobs with the kubernetes-native Spark scheduler:
https://github.com/apache-spark-on-k8s/spark

Using their spark distribution/release, it is so easy to spin up a kubernetes cluster (e.g. in Google GKE) and immediately start running jobs, observing, logs, etc.

For example, submitting a job this way:
```
kubectl proxy &
spark-submit \
  --deploy-mode=cluster \
  --kubernetes-namespace=spark \
  --master=k8s://http://127.0.0.1:8001 \
  ...
 ```

I believe when BigDL initializes (`init_engine`), it fails on validating the `--master` format with `k8s://` in it, not sure how many other things would be involved in adding support for a new scheduler in BigDL, but it would be great to be able to use native kubernetes.

### PR DESCRIPTION
**Pull request by yangw1234**
## What changes were proposed in this pull request?

Support BigDL for Spark on k8s

## How was this patch tested?

Manually tested on both scala and python lenet example

## Related links or issues (optional)
fixed #1679

### CODE REVIEW

File: docs/docs/ProgrammingGuide/run-on-k8s.md [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+
+|Component|Image|
+|---|---|
+|Spark Driver Image|intelanalytics/spark-driver:v2.2.0-kubernetes-0.5.0|
+|Spark Executor Image|intelanalytics/spark-executor:v2.2.0-kubernetes-0.5.0|
+|Spark Initialization Image|intelanalytics/spark-init:v2.2.0-kubernetes-0.5.0|
+|PySpark Driver Image|intelanalytics/spark-driver-py:v2.2.0-kubernetes-0.5.0|
+|PySpark Executor Image|intelanalytics/spark-executor-py:v2.2.0-kubernetes-0.5.0|
+
+You may also build your own customized images.
```
Comment (yangw1234) [inline docs/docs/ProgrammingGuide/run-on-k8s.md]: Should we first publish those docker images to docker hub and add the Docker files to BigDL? @glorysdj

Comment (glorysdj) [reply docs/docs/ProgrammingGuide/run-on-k8s.md]: i will publish docker images and the dockerfiles

Comment (glorysdj) [reply docs/docs/ProgrammingGuide/run-on-k8s.md]: image published to docker hub.
dockerfiles are in PR 2025

**Comment by yangw1234** [comment]
PR validation pass

**Pull request merged**